Remove support for `experimental.otlp` section and `DD_OTLP_*_PORT` env vars

mx-psi opened the pull request:
### What does this PR do?

<!--
* A brief description of the change being made with this pull request.
* If the description here cannot be expressed in a succint form, consider
  opening multiple pull requests instead of a single one.
-->

- Remove support for experimental `experimental.otlp` section.
- Remove support for `DD_OTLP_GRPC_PORT` and `DD_OTLP_HTTP_PORT`

### Motivation

<!--
* What inspired you to submit this pull request?
* Link any related GitHub issues or PRs here.
-->

As announced on the release notes and warning logs, these methods for configuring OTLP ingest are deprecated and were scheduled to be removed on 7.37.

End-users should use the stable `otlp_config` configuration section and the `DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_GRPC_ENDPOINT` and `DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_HTTP_ENDPOINT` environment variables instead.


### Describe how to test/QA your changes

<!--
* Write here in detail or link to detailed instructions on how this change can
  be tested/QAd/validated, including any environment setup.
-->

1. Try to configure OTLP ingest via the experimental section. Check that OTLP is **not** enabled and that there is a warning about an unknown configuration used.
2. Try to configuration OTLP ingest via the `DD_OTLP_GRPC_PORT` and `DD_OTLP_HTTP_PORT` environment variables. Check that OTLP is **not** enabled and that there is a warning about an unknown environment variable being used.

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

mx-psi committed:
[otlp] Remove support for `experimental.otlp` and `DD_OTLP_*_PORT` env vars

mx-psi committed:
[pkg/otlp] Simplify `FromAgentConfig`

mx-psi committed:
[pkg/otlp] Move testcases out of `stable` directory

mx-psi committed:
Add release note

amenasria commented on pkg/config/config_test.go:
```diff
@@ -923,14 +923,6 @@ func TestExperimentalOTLP(t *testing.T) {
 		assert.NoError(t, err)
 		checkConf(t, conf)
 	})
```
Maybe `TestOTLP` would be clearer than `TestExperimentalOTLP`

mx-psi replied:
Good catch! To be honest.. I think we can probably remove this test altogether, since we already test this on the `pkg/otlp/config_test.go` tests.

mx-psi committed:
[pkg/config] Remove `TestExperimentalOTLP` entirely

Unmarshaling of OTLP configuration section is already tested on `pkg/otlp/config_test.go`
